lomiri.*: Fix CMake 4 compat

OPNA2608 opened the pull request:
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 committed:
properties-cpp: 0.0.3 -> 0.0.4

Includes fix for CMake 4 compatibility

OPNA2608 committed:
libqtdbustest: 0.3.2 -> 0.4.0

Includes fix for CMake 4 compatibility

OPNA2608 committed:
lomiri.geonames: Fix CMake 4 compatibility

OPNA2608 committed:
lomiri.lomiri-settings-components: 1.1.2 -> 1.1.3

Includes fix for CMake 4 compatibility

OPNA2608 committed:
process-cpp: Fix CMake 4 compatibility

OPNA2608 committed:
lomiri.lomiri-api: 0.2.2 -> 0.2.3

Includes fix for CMake 4 compatibility

OPNA2608 committed:
lomiri.lomiri-thumbnailer: 3.0.4 -> 3.0.5

Includes fix for CMake 4 compatibility

OPNA2608 committed:
lomiri.gmenuharness: Fix CMake 4 compatibility

OPNA2608 committed:
lomiri.trust-store: Fix CMake 4 compatibility

OPNA2608 committed:
lomiri.qtmir: 0.8.0-unstable-2024-03-06 -> 0.8.0-unstable-2025-05-20

Includes fix for CMake 4 compatibility

OPNA2608 committed:
lomiri.teleports.passthru.tdlib: Fix CMake 4 compatibility

And make it separately buildable.

OPNA2608 committed:
lomiri.lomiri-push-qml: 0-unstable-2022-09-15 -> 0.3.1

Includes fix for CMake 4 compatibility. Also add an updateScript.

OPNA2608 committed:
libsForQt5.quickflux: Fix CMake 4 compatibility

OPNA2608 committed:
lomiri.teleports: Fix CMake 4 compatibility

LordGrimmauld reviewed (changes requested):
Couple notes, but overall very close.

OPNA2608 commented on pkgs/by-name/pr/process-cpp/package.nix:
```diff
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitLab,
+  fetchpatch2,
```
https://github.com/NixOS/nixpkgs/blob/9927c825ded203fcb0c20418ecb9f128871e4076/pkgs/README.md#fetching-patches

> In the interest of keeping our maintenance burden and the size of Nixpkgs to a minimum, patches already merged upstream or published elsewhere *should* be retrieved using `fetchpatch2`:

Aware of the pitfalls, hence usage of `.diff` to get full indices.

> You do have `fetchpatch` in some other changed files...

Predating the change to the guideline / my awareness of the change.

<sup><sub>Not a fan of `fetchpatch2` > `fetchpatch`, but it is what it is…</sub></sup>

LordGrimmauld replied:
Emily is pushing for a revert of that documentation in https://github.com/NixOS/nixpkgs/pull/406131#issuecomment-3288561990, but as long as our documentation says `fetchpatch2` i won't block on this.

OPNA2608 commented on pkgs/desktops/lomiri/development/qtmir/default.nix:
```diff
@@ -37,51 +36,20 @@ stdenv.mkDerivation (finalAttrs: {
-      name = "0001-qtmir-Update-for-Mir-2.15-removals.patch";
-      url = "https://gitlab.com/ubports/development/core/qtmir/-/commit/ead5cacd4d69094ab956627f4dd94ecaff1fd69e.patch";
-      hash = "sha256-hUUUnYwhNH3gm76J21M8gA5okaRd/Go03ZFJ4qn0JUo=";
-    })
```
Not merged into the main branch, but cherry-picked into the branch we're using.

https://gitlab.com/ubports/development/core/qtmir/-/commit/7aab0ad0c6193a761e10ad1c4f65255417524d31

Organisation of stuff like this in QtMir is *reeeeeeally* messy rn, due to Ubuntu Touch using Mir 1.x while we & Debian need Mir 2.x. Will hopefully get cleaned up next year, if they can stick to their roadmap…

OPNA2608 commented on pkgs/desktops/lomiri/development/trust-store/default.nix:
```diff
@@ -56,6 +56,11 @@ stdenv.mkDerivation (finalAttrs: {
+    # Fix compatibility with CMake 4
+    # Remove when https://gitlab.com/ubports/development/core/trust-store/-/merge_requests/20 merged & in release
+    ./1002-3rd_party-xdg-CMakeLists.txt-Bump-minimum-version-to.patch
+    ./1003-CMakeLists.txt-Bump-minimum-version-to-3.10.patch
```
Will switch to fetching.

OPNA2608 replied:
Fetching the fix now.

OPNA2608 commented:
You don't happen to have a full log for the `dbus-cpp` failure, do you? Would love to know if that's fixable, or just something we'll have to accept and disable the flaky test.

OPNA2608 commented:
> I do actually still have the build log! And for once i didn't accidentally override it with a new build... [dbus-cpp-log.txt](https://github.com/user-attachments/files/22921459/dbus-cpp-log.txt)

Thanks!

- `AsyncExecutionLoadTest.RepeatedlyInvokingAnAsyncFunctionWorks` is hammering DBus with 5x100 requests, counts how many succeeded, and hopes to have exactly 500 successful roundtrips within 10 minutes. No reporting on what the error was that a DBus request may've run into, so hard to say why 22 of those requests failed.
  Will keep that suite enabled for now and try to make a patch that prints more useful information, to see if this is just a spurious issue with DBus (and thus not worth blocking the build on) or something that's fixable in the setup.
- `Executor.TimeoutsAreHandledCorrectly` indicates that some memory corruption is happening somewhere (`malloc(): unaligned fastbin chunk detected`). Don't know a fix OTTOMH, and I can't reproduce it.
  I'll report it upstream and disable the affected suite, in hopes that this is an issue specifically with the way this test is written…

OPNA2608 committed:
dbus-cpp: Changes to tests

- Apply a patch that should make a possibly-flaky test more verbose, to ease with debugging.
- Disable the executor test suite, due to a spurious memory corruption in Executor.TimeoutsAreHandledCorrectly.

OPNA2608 commented:
Latest commit should either fix `dbus-cpp`, or make the output abit more verbose to help with debugging.

> Command: `nixpkgs-review pr 450821 --package nixosTests.lomiri`

(If you want to be *super* thorough, some of the packages that are end-user applications also have their own VM tests, like `lomiri.teleports`. But I don't think any of them to have functional changes from any of this.)

LordGrimmauld commented:
I wouldn't be super worried about `dbus-cpp`, it built/tested on the community builder just fine, it might just be my laptop being cursed. If it builds on hydra, it'll be fine.
That said, it now also just succeeded on my laptop, so i am not sure what is going on.

LordGrimmauld reviewed (approved):
thank you!
